modules/wss4j/src/org/apache/ws/security/saml/SAMLIssuerImpl.java

```diff
@@ -148,110 +146,94 @@ public SAMLAssertion newAssertion() { // throws Exception {
-                    .equals(properties.getProperty("org.apache.ws.security.saml.authenticationMethod"))) {
-                authMethod =
-                        SAMLAuthenticationStatement.AuthenticationMethod_Password;
-            }
-            Date authInstant = new Date();
-            Collection bindings = null;
-
-            SAMLSubject subject =
-                    new SAMLSubject(nameId,
-                            Arrays.asList(confirmationMethods),
-                            null,
-                            null);
-            SAMLStatement[] statements =
-                    {
-                        new SAMLAuthenticationStatement(subject,
-                                authMethod,
-                                authInstant,
-                                subjectIP,
-                                null,
-                                bindings)};
-            sa =
-                    new SAMLAssertion(issuer,
-                            null,
-                            null,
-                            null,
-                            null,
-                            Arrays.asList(statements));
-
-            if (!senderVouches) {
-                KeyInfo ki = new KeyInfo(instanceDoc);
-                try {
-                    X509Certificate[] certs =
-                            userCrypto.getCertificates(username);
-                    if (sendKeyValue) {
-                        PublicKey key = certs[0].getPublicKey();
-                        String pubKeyAlgo = key.getAlgorithm();
-                        
-                        if ("DSA".equalsIgnoreCase(pubKeyAlgo)) {
-                            DSAKeyValue dsaKeyValue = new DSAKeyValue(instanceDoc, key);
-                            ki.add(dsaKeyValue);
-                        } else if ("RSA".equalsIgnoreCase(pubKeyAlgo)) {
-                            RSAKeyValue rsaKeyValue = new RSAKeyValue(instanceDoc, key);
-                            ki.add(rsaKeyValue);
-                        }
-                    } else {
-                        X509Data certElem = new X509Data(instanceDoc);
-                        certElem.addCertificate(certs[0]);
-                        ki.add(certElem);
-                    }
-                } catch (WSSecurityException ex) {
-                    if (log.isDebugEnabled()) {
-                        log.debug(ex.getMessage(), ex);
-                    }
-                    return null;
-                } catch (XMLSecurityException ex) {
-                    if (log.isDebugEnabled()) {
-                        log.debug(ex.getMessage(), ex);
+        NameIdentifier nameId = SAMLUtil.newSamlObject(NameIdentifier.class);
+        nameId.setValue(name);
+        nameId.setNameQualifier(qualifier);
+
+        String subjectIP = null;
+        String authMethod = null;
+        if ("password"
+                .equals(properties.getProperty("org.apache.ws.security.saml.authenticationMethod"))) {
```

Comment:
define constants for these property keys

Reply:
fixed in 2f696f6746f53ec59ed149ea7753dbcb09f7b044